### PR TITLE
Improve encounters UI 

### PR DESCRIPTION
**Pull request by brmscheiner**
image
After an individual is assigned show that individual's name instead of "Animal X" (and link to that animals profile). Change button menu content as-per conversation with tanya

### CODE REVIEW

File: src/pages/sighting/encounters/Encounters.jsx [hunk 1]
```diff
@@ -210,6 +213,39 @@ export default function Encounters({
             value: schema.getValue(schema, encounter),
           }),
         );
+
+        const individualGuid = encounter?.individual?.guid;
+        const individualName = deriveIndividualName(
+          encounter?.individual,
+        );
```
Comment (Emily-Ke) [inline src/pages/sighting/encounters/Encounters.jsx:220]: Should this have a fallback? If an identified individual's name can't be derived, there is no text that links to that individual's page.

Comment (Atticus29) [reply src/pages/sighting/encounters/Encounters.jsx:220]: Oh wait. I don't think you can create and individual without giving it a name. I think the form validation prevents you from submitting. So maybe a moot point.

File: src/pages/sighting/encounters/Encounters.jsx [hunk 2]
```diff
@@ -222,52 +258,43 @@ export default function Encounters({
...
-                      id: 'identify-manually',
-                      label: 'Manually assign',
-                      onClick: () =>
-                        setEncounterToAssign(encounterId),
-                    },
-                  ]}
+                  actions={
+                    individualGuid
+                      ? actionButtonActions
+                      : identifyButtonActions
```
[the remark this answers is not in the document]
Comment (Atticus29) [reply src/pages/sighting/encounters/Encounters.jsx:287]: In other words, presence/absences of individual GUID should not be the determining factor of whether we display start identification vs. re-run identification.

Also, does this mean that there will be no way to manually re-assign and individual once it exists?

File: src/components/ButtonMenu.jsx [hunk 3]
```diff
@@ -41,7 +43,11 @@ export default function ButtonMenu({
               handleClose();
             }}
           >
-            {action.label}
+            {action?.labelId ? (
+              <FormattedMessage id={action.labelId} />
+            ) : (
+              action.label
```
[the remark this answers is not in the document]
Comment (brmscheiner) [reply src/components/ButtonMenu.jsx:49]: In this case yes, action will always exist and label will always exist unless labelId does

File: src/pages/sighting/encounters/Encounters.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (brmscheiner) [reply src/pages/sighting/encounters/Encounters.jsx:287]: If an individual guid exists, it means the encounter has already been assigned to an individual. Dunno about re-running ID but we could always add that button later

File: src/pages/sighting/encounters/Encounters.jsx (hunk 1 above)
Comment (Emily-Ke) [reply src/pages/sighting/encounters/Encounters.jsx:220]: I agree that it should not be possible to create an individual from the front end without giving it a name. I think that some migrated zebras do not have first names though.

File: src/pages/sighting/encounters/Encounters.jsx [hunk 4]
```diff
@@ -222,52 +258,43 @@ export default function Encounters({
...
             >
-              <Text
-                id="ANIMAL_CLUSTER_LABEL"
-                variant="h5"
-                values={{ i: i + 1 }}
-                style={{ marginRight: 20 }}
-              />
+              {individualGuid ? (
+                <Link
+                  newTab
```
[the remark this answers is not in the document]
Comment (brmscheiner) [reply src/pages/sighting/encounters/Encounters.jsx:265]: fixed in `Link.jsx`

File: src/pages/sighting/encounters/Encounters.jsx [hunk 5]
```diff
@@ -217,6 +217,8 @@ export default function Encounters({
         const individualGuid = encounter?.individual?.guid;
         const individualName = deriveIndividualName(
           encounter?.individual,
+          'FirstName',
+          'Unnamed Individual'
```
Comment (Atticus29) [inline src/pages/sighting/encounters/Encounters.jsx:221]: We should i18n this at some point, but in no way urgent!

**Review by Emily-Ke — APPROVED** [review]

**Pull request merged**